for-demos/serve_hostname/serve_hostname.go

```diff
@@ -88,8 +90,16 @@ func main() {
 			log.Printf("HTTP request from %s", r.RemoteAddr)
 			fmt.Fprintf(w, "%s", hostname)
 		})
-		go log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", *port), nil))
+		go func() {
+			log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", *port), nil))
```

Comment:
When go log.Fatal(http.ListenAndServe...), ListenAndServe is invoked synchronously (blocking) so the signal code is never reached.  The closure makes the whole thing async, which I am guessing was the origin intent.  I can add this as a comment.